linked_list/main.py

```diff
@@ -62,34 +66,106 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        # start at none
+        self.start = None
+        self.end = None
+        # set elements to empty list
+        elements = [] if not elements else elements
+        for val in elements:
+            self.append(val)
+            
 
     def __str__(self):
-        pass
-
+        return str([element.elem for element in self])
+    
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        n = self.start
+        while n:
+            yield n
+            n = n.next
 
     def __getitem__(self, index):
-        pass
+       for n, node in enumerate(self): 
+           if n == index:
+               return node
 
     def __add__(self, other):
-        pass
+       l1 = [ n.elem for n in self]
+       l2 = [ n.elem for n in other]
+       l3 = l1+l2
+       return LinkedList(l3)
 
     def __iadd__(self, other):
-        pass
+        return self.__add__(other) #  x += 1  == x = x+1
 
     def __eq__(self, other):
-        pass
+        return [element.elem for element in self] == [element.elem for element in other]
 
     def append(self, elem):
-        pass
-
+        #new node
+        if self.start == None:  #is it the first node
+            self.start = Node(elem)
+            self.end = Node(elem)
+        elif self.start == self.end:
```

Comment:
You want to check if your list only has one element right?
The way you do it is not very secure, it will use your equal method defined in the Node class. 
So your condition will be true for a list like [2, 2] (start and end have the same value, but aren't the same nodes).
Use 'is' if you want to check if two variables refer to the same object.